ProjectManagement.BusinessLogic.Tests/Service/BoardServiceTests.cs

```diff
@@ -0,0 +1,546 @@
+            };
+            var identity = new ClaimsIdentity(claims, "basic");
+            var principal = new ClaimsPrincipal(identity);
+            var httpContext = new DefaultHttpContext()
+            {
+                User = principal
+            };
+            mockHttpContextAccessor.Setup(_ => _.HttpContext).Returns(httpContext);
+            _userManager = new UserMananger(mockHttpContextAccessor.Object, _userRepositoryMock.Object);
+            _service = new BoardService(_boardRepositoryMock.Object, _boardMemberRepositoryMock.Object, _userManager);
+        }
+
+        [Fact]
+        public async void GetBoardsAsync_Passed()
+        {
+            //Arrange
+            SetupGetBoards(GetBoards());
+            //Act
+            IEnumerable<Board> boards = await _service.GetBoardsAsync();
+            //Assert
+            Assert.Equal(3, boards.Count());
+        }
+        [Fact]
+        public async void GetBoardAsync_Passed()
+        {
+            //Arrange
+            int boardId = 1;
+            string boardName = "Board 1";
+            List<BoardMember> boardMembers = new List<BoardMember>();
+            BoardMember boardMember = new BoardMember() { UserId = 1 };
+
+            boardMembers.Add(boardMember);
+            var item = new Board
+            {
+                Id = boardId,
+                Name = boardName,
+                BoardMembers = boardMembers
+            };
+            SetupGetBoardById(item);
+            _boardMemberRepositoryMock.Setup(x => x.GetSingleAsync(It.IsAny<GetBoardMemberByUserIdSpecification>())).Returns(Task.FromResult(boardMember));
+            _boardRepositoryMock.Setup(x => x.GetWithItemsByIdAsync(It.IsAny<int>())).Returns(Task.FromResult(item));
+            //Act
+            Board board = await _service.GetBoardByIdAsync(1);
+            //Assert
+            Assert.Equal(item, board);
+        }
+        [Fact]
+        public async void GetBoardAsync_UserIsNotAnMemberOfBoard_ExceptionIsThrown()
+        {
+            //Arrange
+            int boardId = 1;
+            string boardName = "Board 1";
+            BoardMember boardMember = new BoardMember() { UserId = 2 };
+            List<BoardMember> boardMembers = new List<BoardMember>();
+            boardMembers.Add(boardMember);
+            var item = new Board
+            {
+                Id = boardId,
+                Name = boardName,
+                BoardMembers = boardMembers
+            };
+            SetupGetBoardById(item);
+            var memberSpec = new GetBoardMemberByUserIdSpecification(1, boardId);
+            _boardMemberRepositoryMock.Setup(x => x.GetSingleAsync(memberSpec)).Returns(Task.FromResult(boardMember));
+            _boardRepositoryMock.Setup(x => x.GetWithItemsByIdAsync(1)).Returns(Task.FromResult(item));
+
+            //Act
+            //Assert
+
+            await Assert.ThrowsAsync<WebAppException>(async () => await _service.GetBoardByIdAsync(1));
```

Comment:
It is also a good practice to use `Mock.Verify` to ensure that some actions are not executed (e.g. `repository.Add()` is not called)